### PR TITLE
add simple audio occlusion

### PR DESCRIPTION
**Pull request by davidkline-ms**
adds:
* IAudioInfluencer.cs - interface defining methods to implement to become an audio influencer
* AudioEmitter.cs - find and add/remove audio influencers onto an audio source
* AudioOccluder.cs - implementation of IAudioInfluencer providing occlusion

### CODE REVIEW

File: Assets/HoloToolkit/SpatialSound/Scripts/AudioEmitter.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        [Tooltip("Time, in seconds, between audio influence updates.  0 indicates to update every frame.")]
+        [Range(0.0f, 1.0f)]
+        public float UpdateInterval = 0.25f;
+
+        [Tooltip("Maximum distance, in m, to look when attempting to find the user and any influencers.")]
+        [Range(1.0f, 50.0f)]
+        public float MaxDistance = 20.0f;
+
+        [Tooltip("Maximum number of objects that will be considered when looking for influencers.")]
+        [Range(5, 25)]
```
Comment (darax) [inline Assets/HoloToolkit/SpatialSound/Scripts/AudioEmitter.cs]: why is the bottom of the range 5 instead of... 1 or zero?

Comment (davidkline-ms) [inline Assets/HoloToolkit/SpatialSound/Scripts/AudioEmitter.cs]: good question. 1 seems reasonable

File: Assets/HoloToolkit/SpatialSound/Scripts/AudioOccluder.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                lowPass = emitter.AddComponent<AudioLowPassFilter>();
+            }
+            lowPass.enabled = true;
+
+            // In the real world, chaining multiple low-pass filters will result in the 
+            // lowest of the cutoff frequencies being the highest pitches heard.
+            lowPass.cutoffFrequency = Mathf.Min(lowPass.cutoffFrequency, CutoffFrequency);
+
+            // Unlike the cutoff frequency, volume pass-through is cumulative.
+            audioSource.volume *= VolumePassThrough;
```
Comment (darax) [inline Assets/HoloToolkit/SpatialSound/Scripts/AudioOccluder.cs]: how does this get undone in removeeffect?

Comment (darax) [inline Assets/HoloToolkit/SpatialSound/Scripts/AudioOccluder.cs]: okay, I see.

Comment (davidkline-ms) [inline Assets/HoloToolkit/SpatialSound/Scripts/AudioOccluder.cs]: yeah, AudioEmitter.cs is responsible for restoring the volume. 

from Update()
                audioSource.volume = initialAudioSourceVolume;

File: Assets/HoloToolkit/SpatialSound/Scripts/AudioOccluder.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        /// <summary>
+        /// Frequency above the nominal range of human hearing. Applying this frequency to the filter will have no perceived impact on the audio source.
+        /// </summary>
+        private readonly float NeutralFrequency = 22000f;
+
+        [Tooltip("Frequency above which sound will not be heard.")]
+        [Range(10.0f, 22000.0f)]
+        public Single CutoffFrequency = 5000.0f;
+        
+        [Toolip("Percentage of the audio source volume that will be heard after applying occlusion.")]
```
Comment (darax) [inline Assets/HoloToolkit/SpatialSound/Scripts/AudioOccluder.cs]: Toolips are nice, but ToolTips are more practical.

Comment (darax) [inline Assets/HoloToolkit/SpatialSound/Scripts/AudioOccluder.cs]: Please use float.

File: ProjectSettings/EditorBuildSettings.asset [hunk 4]
```diff
@@ -4,4 +4,6 @@
 EditorBuildSettings:
   m_ObjectHideFlags: 0
   serializedVersion: 2
-  m_Scenes: []
+  m_Scenes:
```
Comment (HodgsonSDAS) [inline ProjectSettings/EditorBuildSettings.asset]: Might want to revert the EditorBuildSettings.asset

File: Assets/HoloToolkit/SpatialSound/Scripts/IAudioInfluencer.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    /// Interface that is implemented by any class that wishes to influence how an audio source sounds.
+    /// </summary>
+    public interface IAudioInfluencer
+    {
+        /// <summary>
+        /// Applies an audio effect.
+        /// </summary>
+        /// <param name="emitter">The GameObject on which the effect is to be applied.</param>
+        /// <param name="audioSource">The AudioSource that will be impacted by the effect.</param>
+        void ApplyEffect(GameObject emitter,
```
Comment (NeerajW) [inline Assets/HoloToolkit/SpatialSound/Scripts/IAudioInfluencer.cs]: Hmm given that the signatures are same I would have expected ApplyEffect to perhaps return the effect GO so I don't have to do a GetComponent again. And that effect passed into RemoveEffect. But I guess this works too.

Comment (davidkline-ms) [inline Assets/HoloToolkit/SpatialSound/Scripts/IAudioInfluencer.cs]: There is no requirement for an influencer to use a single filter. Returning a GameObject would be insufficient.

**Pull request merged**